cheroot/server.py

```diff
@@ -1175,6 +1175,12 @@ def send_headers(self):
                 if not self.close_connection:
                     self.outheaders.append((b'Connection', b'Keep-Alive'))
 
+        if dict(self.outheaders).get(b'Connection', b'') == b'Keep-Alive':
```

Comment:
I converting to dict would probably be more expensive than having a lookup helper function that would exit early. @meaksh WDYT?

Reply:
Or maybe this would be a bit faster
```suggestion
        if (b'Connection', b'Keep-Alive') in set(self.outheaders):
```

Reply:
Yeah, I think we could simply iterate the `self.outheaders` list and do tuple comparison without any casting - I did some tests and that seems to be the faster way.